go.mod

```diff
@@ -1,10 +1,11 @@
 module github.com/signadot/cli
 
-go 1.23
+go 1.23.0
```

Comment:
If you keep this as 1.23 then it allows us to upgrade minor versions of Go more easily.

Reply:
go mod tidy is doing this looks like - is there a way you're aware of that prevents this?

Reply:
arg they changed it so it overwrites that now by default.

I'm ok with removing this discussion from this PR, but I think we need to either edit the go.mod manually after go mod tidy or explicitly track minor version upgrades.